MIP/mip-58/README.md

```diff
@@ -0,0 +1,462 @@
+# MIP-58: Trusted-Relayer focused Bridge Design
+
+- **Description**: Proposes a bridge design that capitalizes on the trust-assumption on the relayer, reduces the number of messages, and potentially cost.
+- **Authors**: [Primata](mailto:primata@movementlabs.xyz)
+- **Desiderata**: [MD-21](../MD/md-21)
+
+## Abstract
+
+This proposal advocates for replacing the current HTLC-based bridge design, which requires four transactions and introduces significant complexity, with a simpler and more efficient two-transaction mechanism. The proposed design reduces costs, enhances security, minimizes user friction, and avoids potential exploits caused by refund logic. It addresses long-standing issues such as reliance on sponsored transactions, outdated audits, and unimplemented fee mechanisms speeding the process to achieve a fully functional bridge that movement yet has been unable to complete, and lots of under issues raised and still pending.
+
+## Motivation
+
+The current bridge design poses numerous challenges, including inefficiency, cost, and user frustration:
+
+1. **Transaction Complexity**: Requires four interactions:
+   - Two by the user (initiation and finalization).
+   - One by the relayer to complete the transaction.
+   - One by a third party (tasked by the relayer) to finalize refunds in case of failure.
+   - If any of these fails, users and the protocol face losses.
+2. **High Cost**:
+   - The multi-transaction setup is expensive for both users and the protocol.
+   - Misestimations in fee calculations could result in significant losses.
+   - Example: At 16 gwei, bridging costs $20 for 10k transactions. If fees are miscalculated by 25%, this leads to a $25k loss. At 100 gwei, the loss escalates dramatically.
+3. **Unfriendly User Experience**:
+   - The current design requires users to have funds on the destination chain for finalization. This increases friction and discourages adoption.
+   - Sponsored transactions are essential to the current implementation but remain unimplemented, leaving users stranded without funds.
+4. **Security Risks**:
+   - Relayer keys are a critical point of failure. This assumption cannot be avoided.
+   - Refund logic introduces vulnerabilities where attackers could exploit the rate limit and relayer downtime.
+   - For instance, if the relayer fails to finalize on the initiator, a malicious actor could:
+     - Take over the refund keys.
+     - Exploit both directions of the bridge (e.g., ETH→MOVE and MOVE→ETH) repeatedly, draining funds.
+5. **Maintenance Burden**:
+   - The current audit is outdated and does not reflect the significant changes made since.
+   - The current design requires a complete UI/UX overhaul, adding complexity and delay.
+6. **Unnecessary Complexity**:
+   - HTLC-based bridges are largely abandoned in favor of simpler, more effective designs.
```

Comment:
HTLC is used for _atomic swaps_. [Chainlink](https://chain.link/education-hub/atomic-swaps) has an explainer for atomic swaps. 
HTLC-bridges are used to implement atomic swaps between parties that do not trust each other. 
HTLC is not _largely_ abandoned (see the link above). 
The problem is that we have used an atomic swap architecture [RFC-40](https://github.com/movementlabsxyz/rfcs/blob/main/0040-atomic-bridge/rfc-0040-atomic-bridge.md) to guide the implementation of the bridge. 
But a bridge (transfer) is different to a swap: in a swap parties can withdraw from the deal before the end of an agreed period. They also have to escrow funds.
In a bridge, one party wants to _transfer_ assets from one chain A to another chain B.
Most bridges implement a lock/mint or mint/burn mechanism, not an atomic swap.